### ISSUE TITLE
Not working on Edge

### ISSUE THREAD

**Issue by cryptolfc** [proginosko/LeechBlockNG-edge#10]
Extension isn't working on Edge (Windows). The time pops up sometimes and mostly doesn't appear on screen. Even when it does, there is no countdown. The statistics counter doesn't register any time either but the diagnostic page shows that the site (youtube) is indeed getting blocked. Please help, I need this bad! (can't otherwise concentrate on studies)
Settings:
![image](https://github.com/user-attachments/assets/be324821-15bc-440b-8875-0e4ddce15ce1)
![image](https://github.com/user-attachments/assets/bdb0c707-c85d-460b-914a-727f2aa56407)
![image](https://github.com/user-attachments/assets/50d872f6-2530-4d18-b08d-480b3ce0d0fc)
Diagnostic:
![image](https://github.com/user-attachments/assets/a8bcaaf9-564f-4677-b421-7cb81c995701)
Statistics:
![image](https://github.com/user-attachments/assets/31cd6d6d-38e5-421b-9daf-a4f382a1c1fa)

**Comment by proginosko** [comment]
Go to the Extensions page (edge://extensions). Click on "Details" under LeechBlock NG. Under "Site access" ensure that "On all sites" is selected.

**Comment by cryptolfc** [comment]
It indeed is selected.
![image](https://github.com/user-attachments/assets/ab544f6b-7bf2-45c2-9d0d-514540f8e90d)

**Comment by cryptolfc** [comment]
Update: I have tried removing the timer to block site completely & it works. So that should mean only the timer isn't working?

**Comment by proginosko** [comment]
What do you mean by "removing the timer"? Which setting did you change?

**Comment by cryptolfc** [comment]
"Enter a time limit after which to block these sites (optional):
___ minutes in every _____"
Not setting time limit. 
However, statistics still doesn't update any time.

**Comment by proginosko** [comment]
Ah, I think I see what it is. Deselect the option "Count time spent on these sites only when browser tab is playing audio".

**Comment by cryptolfc** [comment]
Still not working & statistics still not updating. Do I require a restart for my laptop as well? 

[FYI, I have Windows 8.1 with Edge Version 109.0.1518.140]

**Comment by proginosko** [comment]
Is the problem only with youtube.com or does it happen for all websites in a block set?

There have been some reports of the Chrome extension not working properly after an update. You could try exporting your options to a text file, uninstalling and reinstalling the extension, then importing the options again.

**Comment by cryptolfc** [comment]
Tried with other sites - still not working.
Tried uninstalling and reinstalling - not working.
I guess I will have to completely block sites, and use override.

**Comment by proginosko** [comment]
Please export your options to a text file (Options > General > Export Options) and email it to me at leechblockng@proginosko.com so I can take a closer look.

**Comment by proginosko** [comment]
Very strange. I tried your options and they work fine on my system (Edge on Windows 11).

You could try this: Go to the Extensions page (edge://extensions/) and click on the "service worker" link under LeechBlock. A console window should open. Click the "Clear console" icon then visit YouTube in the main browser window. Do any error messages appear in the console?

**Comment by cryptolfc** [comment]
Tried and this too didn't work. Something's very strange about this. Do you reckon it has something to do with an older Operating System or browser?

**Comment by proginosko** [comment]
When you say "this too didn't work" are you referring to the suggestion about opening the console? If so, in what sense did it not work?

I see that you do have an old version of Edge. The current version is 131. I'm not sure whether version 109 supports Manifest Version 3 (MV3) which is what LeechBlock uses.

**Comment by cryptolfc** [comment]
Yes, I cleared the console; opened youtube; Timer appears but doesn't countdown; Statistics still doesn't record any time.

I will update my browser sometime soon and come back to this. Thank you for helping out. :)
